test/Main.hs

```diff
@@ -25,47 +25,74 @@ tests :: TestTree
 tests =
   testGroup
     "tests"
-    [testCase "test PrintConnectionStats" testPrintConnectionStats]
+    [ testCase "test getLastQuery" testGetLastQuery
+    , testCase "test withFrozenLastQuery" testWithFrozenLastQuery
+    , testCase "test connection stats retrieval with new connection" testConnectionStatsWithNewConnection
+    ]
 
-testPrintConnectionStats :: Assertion
-testPrintConnectionStats = do
+testGetLastQuery :: Assertion
+testGetLastQuery = do
   dbUrl <- T.pack <$> getEnv "DATABASE_URL"
   let connectionSource :: ConnectionSource [MonadBase IO, MonadMask]
       connectionSource = simpleSource $ ConnectionSettings dbUrl Nothing []
-      transactionSettings = defaultTransactionSettings
-      sql = "SELECT 1"
-      program :: Eff '[DB, Error HPQTypesError, IOE] ()
-      program = do
-        rowNo <- runQuery $ mkSQL sql
-        liftBase $ putStr "Row number: " >> print rowNo
+  void . runEff . runErrorNoCallStack @HPQTypesError . runEffectDB connectionSource defaultTransactionSettings $ do
+    do
+      -- Run the first query and perform some basic sanity checks
+      let sql = mkSQL "SELECT 1"
+      rowNo <- runQuery sql
+      liftBase $ assertEqual "One row should be retrieved" 1 rowNo
```

Comment:
I see you've changed some `liftIO` calls to `liftBase`, but in the spirit of the above comment about `MonadBase` let's get rid of `liftBase` here completely, please.

Reply:
:+1: